src/librustc/hir/def.rs

```diff
@@ -53,13 +53,13 @@ pub enum Def {
     Existential(DefId),
     /// `type Foo = Bar;`
     TyAlias(DefId),
-    TyForeign(DefId),
+    Foreign(DefId),
```

Comment:
This seems suboptimal... what if you *first* rename this to `ForeignTy` *and then* do the other renames?